tracing-core/src/dispatch.rs

```diff
@@ -185,21 +187,29 @@ thread_local! {
 }
 
 static EXISTS: AtomicBool = AtomicBool::new(false);
-static GLOBAL_INIT: AtomicUsize = AtomicUsize::new(UNINITIALIZED);
+static GLOBAL_STATE: AtomicUsize = AtomicUsize::new(UNINITIALIZED);
 
 #[cfg(feature = "std")]
 static SCOPED_COUNT: AtomicUsize = AtomicUsize::new(0);
 
 const UNINITIALIZED: usize = 0;
 const INITIALIZING: usize = 1;
 const INITIALIZED: usize = 2;
+#[cfg(feature = "std")]
+const IS_GLOBAL_SCOPED: usize = 3;
 
 static mut GLOBAL_DISPATCH: Dispatch = Dispatch {
     #[cfg(feature = "alloc")]
     collector: Kind::Global(&NO_COLLECTOR),
     #[cfg(not(feature = "alloc"))]
     collector: &NO_COLLECTOR,
 };
+
+#[cfg(feature = "std")]
+lazy_static! {
+    static ref GLOBAL_SCOPED: RwLock<Dispatch> = RwLock::new(NONE.clone());
```

Comment:
How about using `AtomicPtr` initialized to `&GLOBAL_DISPATCH`?

Similarly to how `log` crate does its logger, we could have API like `set_scoped_global_logger(dispatch: &'static Dispatch)`

Reply:
hmm, i would prefer to avoid making this API require that the dispatcher be `&'static` --- one of the advantages of a scoped default dispatcher is that the dispatcher itself can be dropped at the end of the program, allowing it to release any resources it holds onto, flush output streams, et cetera. if this API required the dispatcher to live in a static, we would lose the ability to actually drop it.

however, i think we could also handle the `Arc` dispatcher case and still replace the `RwLock` with an `AtomicPtr`, by "leaking" a clone of the `Arc` into the static, as long as we ensured that the leaked arc ref is decremented when the global default is unset...

Reply:
ah but I think it is ok to let user to do cleanup.
In first place, it is not tokio who is interested in it, but user so having user to provide pointer will just shit responsibility to cleanup to user

With reference you could simplify implementation to not rely on locks, but I suppose as long as it is `parking_lot` RwLock it should have minimal impact on performance